src/movies.js

```diff
@@ -1,16 +1,170 @@
 // Iteration 1: All directors? - Get the array of all directors.
+const getAllDirectors = (array) => {
+    
+    let directors = array.map(movie => {
+        return movie.director; 
+    })
+    return directors;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+const howManyMovies = (array) => {
+    let dramaMovies = array.filter((movie) => {
+        return movie.genre.includes("Drama") && movie.director === "Steven Spielberg";
+        // return movieType.genre.includes("drama");
+    })
+    return dramaMovies.length
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+const ratesAverage = (array) => {
+    
+    if (array != "") {
+
+        let averageRates = array.reduce((acc, i) => {
+            if (i.rate) {
+            return acc + i.rate / array.length
```

Comment:
check indentation level here.

Reply:
Very smart using the "/array.length" here instead of after the reduce for the whole sum. This is some math magic that allows this but that way you can reduce the code, very clever :)